lib/install/deps.js

```diff
@@ -613,8 +635,12 @@ var earliestInstallable = exports.earliestInstallable = function (requiredBy, tr
   // if this tree location requested the same module then we KNOW it
```

Comment:
@iarna: While working on this issue, this caveat bothered me somewhat, as it makes `earliestInstallable` highly dependent on the behavior of `findRequirement` (which is particularly troublesome, given that `earliestInstallable` is exported and used elsewhere)

It may be worth considering somehow merging the two methods, or adding some sort of safeguard mechanism so that we can verify that `findRequirement` indeed did its job.  

You seem to know this code better than anyone else -- I'd be interested in hearing your thoughts about how we might be able to make this more robust without hurting performance.